### DISCUSSION TITLE
Dev Environment Setup

### DISCUSSION THREAD

**Discussion by samiam21** [firefly-iii/firefly-iii#5975] [Firefly III]
Hi, I would like to set up a development environment to make some customized features for my own use.  I have never done php/nginx dev before (I've done a lot of .NET, JavaScript, React, etc.).

Could someone point me to some help or instructions on how to set up a development environment so I can run the application locally and make changes to it?  Thanks!

**Comment by JC5** [comment]
There are many ways to set up a dev environment. Personally, I have a vagrant linux (ubuntu) VM with a mapping to my normal PC + nginx. But it also depends on your OS and other factors.

**Reply by samiam21** [reply]
I was originally thinking of trying to use WSL but a VM sounds like a better idea, thanks!
